File: services/moltbot-secure/clients/javascript/src/exceptions.ts
/**
 * Custom exceptions for Moltbot Secure client
 */

export class MoltbotSecureError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "MoltbotSecureError";
    Object.setPrototypeOf(this, MoltbotSecureError.prototype);
  }
}

export class AuthenticationError extends MoltbotSecureError {
  constructor(message: string) {
    super(message);
    this.name = "AuthenticationError";
    Object.setPrototypeOf(this, AuthenticationError.prototype);
  }
}

export class EncryptionError extends MoltbotSecureError {
  constructor(message: string) {
    super(message);
    this.name = "EncryptionError";
    Object.setPrototypeOf(this, EncryptionError.prototype);
  }
}

export class SessionError extends MoltbotSecureError {
  constructor(message: string) {
    super(message);
    this.name = "SessionError";
    Object.setPrototypeOf(this, SessionError.prototype);
  }
}

export class ProxyError extends MoltbotSecureError {
  constructor(message: string) {
    super(message);
    this.name = "ProxyError";
    Object.setPrototypeOf(this, ProxyError.prototype);
  }
}

File: services/moltbot-secure/clients/python/moltbot_secure/exceptions.py
"""
Custom exceptions for Moltbot Secure client
"""


class MoltbotSecureError(Exception):
    """Base exception for Moltbot Secure client errors"""
    pass


class AuthenticationError(MoltbotSecureError):
    """Authentication-related errors"""
    pass


class EncryptionError(MoltbotSecureError):
    """Encryption-related errors"""
    pass


class SessionError(MoltbotSecureError):
    """Session-related errors"""
    pass


class ProxyError(MoltbotSecureError):
    """Proxy communication errors"""
    pass
